File: java/src/main/java/com/example/hexagonal/adapter/outbound/HypotheticalMessageSender.java
package com.example.hexagonal.adapter.outbound;

import com.example.hexagonal.application.port.outbound.MessagePort;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;

@Component
@Slf4j
public class HypotheticalMessageSender implements MessagePort {

  @Override
  public void send(Object aHypotheticalMessage) {
    log.info("Send a message: {}", aHypotheticalMessage);
  }
}

File: kotlin/src/main/kotlin/com/example/hexagonal/adapter/outbound/HypotheticalMessageSender.kt
package com.example.hexagonal.adapter.outbound

import com.example.hexagonal.application.port.outbound.MessagePort
import io.github.oshai.kotlinlogging.KotlinLogging
import org.springframework.stereotype.Component

@Component
class HypotheticalMessageSender : MessagePort {

    val logger = KotlinLogging.logger {}

    override fun send(aHypotheticalMessage: Any) {
        logger.info { "Send a message: $aHypotheticalMessage" }
    }
}
